### PR TITLE
Introduce DOMWorld

### LINKED ISSUE #876: Split out DOMWorld from Frame
https://github.com/kblok/puppeteer/pull/15/commits/4ac00caf9db30b3883e0ed3e5da93160872c4648

### PR DESCRIPTION
**Pull request by Meir017**
closes #876

### CODE REVIEW

**Comment by Meir017** [comment]
@kblok ready

File: lib/PuppeteerSharp/FrameManager.cs [hunk 1]
```diff
@@ -327,8 +327,8 @@ private void OnFrameNavigatedWithinDocument(NavigatedWithinDocumentResponse e)
         private void RemoveContext(ExecutionContext context)
         {
             if (context.Frame != null)
-            {
-                context.Frame.SetDefaultContext(null);
+            {
```
Comment (kblok) [inline lib/PuppeteerSharp/FrameManager.cs]:
```suggestion
            {
```

**Comment by kblok** [comment]
I thought it was going to be more difficult. Good job!

**Comment by Meir017** [comment]
there is still one failing test, not sure how to fix it 🤔

**Comment by kblok** [comment]
I think you started moving the code before this https://github.com/kblok/puppeteer-sharp/pull/907/files#diff-cf4a9fad47c66b5c54f7f253f685b7a6L567

We are losing that change.

**Pull request closed without merging**